src/packages/contrib/contrib.cc

```diff
@@ -1610,116 +1610,131 @@ void f_replaceable(void) {
-        ignore[i + 2] = 0;
-      }
-    }
-    numignore += 2;
     obj = (sp - 1)->u.ob;
-  } else {
-    numignore = 2;
-    ignore = reinterpret_cast<char **>(DCALLOC(2, sizeof(char *), TAG_TEMPORARY, "replaceable"));
-    ignore[0] = findstring(APPLY_CREATE);
-    ignore[1] = findstring(APPLY___INIT);
+  }
+  else {
     obj = sp->u.ob;
   }
 
   prog = obj->prog;
-  num = prog->num_functions_defined + prog->last_inherited;
-
-  for (i = 0; i < num; i++) {
-    if (prog->function_flags[i] & (FUNC_INHERITED | FUNC_NO_CODE)) {
-      continue;
+  if (obj == simul_efun_ob || prog->func_ref) {
+    replaceable = 0;
+  }
+  else {
+    if (st_num_arg == 2) {
+      numignore = sp->u.arr->size;
+      if (numignore) {
+        ignore = reinterpret_cast<char **>(
+            DCALLOC(numignore + 2, sizeof(char *), TAG_TEMPORARY, "replaceable"));
+      } else {
+        ignore = 0;
+      }
+      ignore[0] = findstring(APPLY_CREATE);
+      ignore[1] = findstring(APPLY___INIT);
+      for (i = 0; i < numignore; i++) {
+        if (sp->u.arr->item[i].type == T_STRING) {
+          ignore[i + 2] = findstring(sp->u.arr->item[i].u.string);
+        } else {
+          ignore[i + 2] = 0;
+        }
+      }
+      numignore += 2;
+    } else {
+      numignore = 2;
+      ignore = reinterpret_cast<char **>(DCALLOC(2, sizeof(char *), TAG_TEMPORARY, "replaceable"));
+      ignore[0] = findstring(APPLY_CREATE);
+      ignore[1] = findstring(APPLY___INIT);
     }
-    for (j = 0; j < numignore; j++) {
-      if (ignore[j] == find_func_entry(prog, i)->funcname) {
+
+    num = prog->num_functions_defined + prog->last_inherited;
+
+    for (i = 0; i < num; i++) {
+      if (prog->function_flags[i] & (FUNC_INHERITED | FUNC_NO_CODE)) {
+        continue;
+      }
+      for (j = 0; j < numignore; j++) {
+        if (ignore[j] == find_func_entry(prog, i)->funcname) {
+          break;
+        }
+      }
+      if (j == numignore) {
         break;
       }
     }
-    if (j == numignore) {
-      break;
-    }
-  }
 
-  replaceable = (i == num);
-  if (obj == simul_efun_ob || prog->func_ref) {
-    replaceable = 0;
+    replaceable = (i == num);
+    FREE(ignore);
   }
 
   if (st_num_arg == 2) {
     free_array((sp--)->u.arr);
   }
-  FREE(ignore);
   free_svalue(sp, "f_replaceable");
   put_number(replaceable);
 }
 #endif
 
 #ifdef F_PROGRAM_INFO
+typedef struct {
+    int h_c,    // how many headers?
+        p_s,    // how big are the compiled programs?
+        ff_c,   // how many functions flags?
+        fd_c,   // how many function definitions?
+        s_c,    // how many strings?
+        v_c,    // how many variables?
+        i_c,    // how many inherits?
+        nc_c,   // how many classes (structs)?
+        mc_c,   // how many class members?
+        t_c,    // how mamy saved types (#pragma save_types)?
+        total;  // how big all together?
+} program_info_t;
+
+inline void sum_program_info(const program_t * const p, program_info_t &info) {
```

Comment:
RVO?

If we where to return the struct it would be come **much** slower since it would be necessary to copy the whole structure many times:
```c++
for(ob = obj_list; ob; ob = ob->next_all) {
    if(…) { continue; }
    info = sum_program_info(ob->prog, info);
}
```
The way it's currently used only a single pointer needs to be transfered, if the function realy is inlined not even that…